docs(developer.md): add note about ulimit workaround

b5 opened the pull request:
came up a few times in discussion on discord, might as well have a linkable header we can point to

Mr0grog commented on DEVELOPERS.md:
```diff
@@ -166,4 +165,18 @@ A detailed explanation can be found in this [document][commit-message-format].
 
+## <a name="troubleshooting"> Troubleshooting
+
+#### "Too Many Open Files" on OS X
```
Pedantic fix (not that anybody is confused by this, though):

```suggestion
#### "Too Many Open Files" on MacOS
```

Mr0grog commented on DEVELOPERS.md:
```diff
@@ -166,4 +165,18 @@ A detailed explanation can be found in this [document][commit-message-format].
+
+#### "Too Many Open Files" on OS X
+
+If while running the test suite you get failing tests with "too many open files", you may need to adjust the number of files your shell process is willing to open. This is particularly common with the `p2p` and `cmd` pacakges. Both rely heavily on interacting temporary directories on the file system to run tests.
```
```suggestion
If, while running the test suite, you get failing tests with "too many open files", you may need to adjust the number of files your shell process is willing to open. This is particularly common with the `p2p` and `cmd` pacakges. Both rely heavily on interacting temporary directories on the file system to run tests.
```

Mr0grog commented on DEVELOPERS.md:
```diff
@@ -166,4 +165,18 @@ A detailed explanation can be found in this [document][commit-message-format].
+
+If while running the test suite you get failing tests with "too many open files", you may need to adjust the number of files your shell process is willing to open. This is particularly common with the `p2p` and `cmd` pacakges. Both rely heavily on interacting temporary directories on the file system to run tests.
+
+Often this is caused by having too low a `ulimit` value for the shell. Try the following command:
```
```suggestion
Often this is caused by having too low a file limit for your shell. You can use the [`ulimit` command](https://ss64.com/osx/ulimit.html) to check or change the limits. Try the following command to set the limit to 1000 open files:
```

Mr0grog commented on DEVELOPERS.md:
````diff
@@ -166,4 +165,18 @@ A detailed explanation can be found in this [document][commit-message-format].
+ulimit -S -n 1000
+```
+
+We recommend a ulimit value of at least 1000, feel free to go higher if you need. Some other software require values as high as 10000 to avoid max open file issue.
````
```suggestion
We recommend a ulimit value of at least 1000; feel free to go higher if you need. Some other software require values as high as 10000 to avoid the max open file issue.
```

b5 commented:
Thanks so much for the detailed review @Mr0grog. I've incorporated all of your fixes, learned to refer to it as `MacOS`, and rebased to incorporate fixes from #1174.
